pkg/claimsstore/util.go

```diff
@@ -2,10 +2,72 @@ package claimsstore
 
 import (
 	"bytes"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/google/uuid"
+	didlib "github.com/ockam-network/did"
+)
+
+var (
+	// PrefixRootMerkleTree prefix value for root tree
+	PrefixRootMerkleTree = []byte("root_merkletree")
+	// ErrTooLongDIDMethod did to binary is a bit fragile for now, fails when method is more than 15 bytes
+	ErrTooLongDIDMethod = errors.New("method string is too long to fit in merkletree elembytes")
+	// ErrWrongSizByteSliceDID if the slice is the wrong size can't conver it back to a did
+	ErrWrongSizByteSliceDID = errors.New("binaryToDID expects a byte slice of length 32")
+	singleEmptyByte         = []byte{32}
 )
 
-// internal method from iden3 db that seemed necessary to implement the interface
-func concat(vs ...[]byte) []byte {
+// DIDToBinary converts did into []bytes with 16 bytes as the uuid and 16 bytes as method
+func DIDToBinary(did *didlib.DID) ([]byte, error) {
+	uid, err := uuid.Parse(did.ID)
```

Comment:
We can prob punt on this, but if we support DIDs outside of our own, we would need to support DID IDs that are not UUIDs.  According to the spec, DID method IDs just need to be a set of characters https://w3c-ccg.github.io/did-spec/#generic-did-syntax